esp/src/eclwatch/HPCCPlatformWidget.js

```diff
@@ -418,17 +418,14 @@ define([
                     handleAs: "text"
                 }).then(function (response) {
                     context.configText = context.formatXml(response);
-                    context.configSourceCM = CodeMirror.fromTextArea(dom.byId(context.id + "ConfigTextArea"), {
-                        tabMode: "indent",
-                        matchBrackets: true,
-                        lineNumbers: true,
-                        mode: "xml",
-                        readOnly: true,
-                        foldGutter: true,
-                        gutters: ["CodeMirror-linenumbers", "CodeMirror-foldgutter"]
-                    });
-                    context.configSourceCM.setSize("100%", "100%");
-                    context.configSourceCM.setValue(context.configText);
+                    context.configSourceCM = new CodeMirror.XMLEditor();
+
+                    var t = window.setTimeout(() => {
+                        context.configSourceCM.target(dom.byId(context.id + "_Config")).render();
+                        dom.byId(context.id + "ConfigTextArea").style.display = "none";
+                        window.clearTimeout(t);
```

Comment:
Is clear timeout really needed?

Reply:
Possibly not. This was a pattern I'd gotten into some years ago, to always unset a timeout to avoid leaks. Not sure how that would translate into the React paradigm. Maybe something in a hook that for component unmounting.